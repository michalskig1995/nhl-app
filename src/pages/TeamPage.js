import React, { useState, useEffect } from "react";
import { useParams, Link } from "react-router-dom";

import { getNHLTeams } from "../api/api";

const TeamPage = () => {
  const { teamId } = useParams();
  const [error, setError] = useState(null);
  const [isLoaded, setIsLoaded] = useState(false);
  const [team, setTeam] = useState(null);

  useEffect(() => {
    getNHLTeams().then(
      result => {
        setIsLoaded(true);
        const teamById = result.teams.find(t => t.id == teamId);
        if (teamById) {
          setTeam(teamById);
        }
      },

      error => {
        setIsLoaded(true);
        setError(error);
      }
    );
  }, []);

  if (error) {
    return <div>Error: {error.message}</div>;
  } else if (!isLoaded) {
    return <div>Loading</div>;
  } else {
    return (
      <>
        <h1>{team.name}</h1>
        <ul>
          <li>Conference: {team.conference.name}</li>
          <li>Division: {team.division.name}</li>
          <li>
            Roster
            <ul>
              {team.roster.roster.map(person => (
                <li key={person.person.id}>
                  <Link to={`/players/${person.person.id}`}>
                    {person.person.fullName}
                  </Link>
                </li>
              ))}
            </ul>
          </li>
        </ul>
      </>
    );
  }
};

export default TeamPage;
