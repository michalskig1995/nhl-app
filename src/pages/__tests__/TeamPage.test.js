import React from "react";
import { BrowserRouter, Routes, Route } from "react-router-dom";

import { render, screen, waitFor } from "@testing-library/react";
import TeamPage from "../TeamPage";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useParams: () => ({
    teamId: 1
  })
}));

describe("TeamPage", () => {
  beforeEach(() => jest.clearAllMocks());

  it("renders the NHL TeamPage from NHL teams api for a given team", async () => {
    const wrapper = render(
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<TeamPage />} />
        </Routes>
      </BrowserRouter>
    );
    await waitFor(() => {
      // Ensure page has loaded
      screen.getByText("New Jersey Devils");
      expect(wrapper).toMatchSnapshot();
    });
  });
});
