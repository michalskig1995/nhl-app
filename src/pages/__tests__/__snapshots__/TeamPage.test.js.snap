// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamPage renders the NHL TeamPage from NHL teams api for a given team 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1>
        New Jersey Devils
      </h1>
      <ul>
        <li>
          Conference: 
          Eastern
        </li>
        <li>
          Division: 
          Metropolitan
        </li>
        <li>
          Roster
          <ul>
            <li>
              <a
                href="/players/8473541"
              >
                Jonathan Bernier
              </a>
            </li>
            <li>
              <a
                href="/players/8476292"
              >
                Ondrej Palat
              </a>
            </li>
            <li>
              <a
                href="/players/8478406"
              >
                Mackenzie Blackwood
              </a>
            </li>
            <li>
              <a
                href="/players/8482076"
              >
                Nico Daws
              </a>
            </li>
            <li>
              <a
                href="/players/8474090"
              >
                Brendan Smith
              </a>
            </li>
            <li>
              <a
                href="/players/8475193"
              >
                Tomas Tatar
              </a>
            </li>
            <li>
              <a
                href="/players/8475287"
              >
                Erik Haula
              </a>
            </li>
            <li>
              <a
                href="/players/8476462"
              >
                Dougie Hamilton
              </a>
            </li>
            <li>
              <a
                href="/players/8476923"
              >
                Damon Severson
              </a>
            </li>
            <li>
              <a
                href="/players/8477341"
              >
                Andreas Johnsson
              </a>
            </li>
            <li>
              <a
                href="/players/8477425"
              >
                Miles Wood
              </a>
            </li>
            <li>
              <a
                href="/players/8477435"
              >
                Ryan Graves
              </a>
            </li>
            <li>
              <a
                href="/players/8477970"
              >
                Vitek Vanecek
              </a>
            </li>
            <li>
              <a
                href="/players/8478399"
              >
                Jonas Siegenthaler
              </a>
            </li>
            <li>
              <a
                href="/players/8478507"
              >
                John Marino
              </a>
            </li>
            <li>
              <a
                href="/players/8479407"
              >
                Jesper Bratt
              </a>
            </li>
            <li>
              <a
                href="/players/8479414"
              >
                Nathan Bastian
              </a>
            </li>
            <li>
              <a
                href="/players/8479415"
              >
                Michael McLeod
              </a>
            </li>
            <li>
              <a
                href="/players/8480002"
              >
                Nico Hischier
              </a>
            </li>
            <li>
              <a
                href="/players/8480003"
              >
                Jesper Boqvist
              </a>
            </li>
            <li>
              <a
                href="/players/8480188"
              >
                Fabian Zetterlund
              </a>
            </li>
            <li>
              <a
                href="/players/8480860"
              >
                Kevin Bahl
              </a>
            </li>
            <li>
              <a
                href="/players/8481033"
              >
                Akira Schmid
              </a>
            </li>
            <li>
              <a
                href="/players/8481068"
              >
                Yegor Sharangovich
              </a>
            </li>
            <li>
              <a
                href="/players/8481559"
              >
                Jack Hughes
              </a>
            </li>
            <li>
              <a
                href="/players/8482110"
              >
                Dawson Mercer
              </a>
            </li>
            <li>
              <a
                href="/players/8482125"
              >
                Alexander Holtz
              </a>
            </li>
          </ul>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <h1>
      New Jersey Devils
    </h1>
    <ul>
      <li>
        Conference: 
        Eastern
      </li>
      <li>
        Division: 
        Metropolitan
      </li>
      <li>
        Roster
        <ul>
          <li>
            <a
              href="/players/8473541"
            >
              Jonathan Bernier
            </a>
          </li>
          <li>
            <a
              href="/players/8476292"
            >
              Ondrej Palat
            </a>
          </li>
          <li>
            <a
              href="/players/8478406"
            >
              Mackenzie Blackwood
            </a>
          </li>
          <li>
            <a
              href="/players/8482076"
            >
              Nico Daws
            </a>
          </li>
          <li>
            <a
              href="/players/8474090"
            >
              Brendan Smith
            </a>
          </li>
          <li>
            <a
              href="/players/8475193"
            >
              Tomas Tatar
            </a>
          </li>
          <li>
            <a
              href="/players/8475287"
            >
              Erik Haula
            </a>
          </li>
          <li>
            <a
              href="/players/8476462"
            >
              Dougie Hamilton
            </a>
          </li>
          <li>
            <a
              href="/players/8476923"
            >
              Damon Severson
            </a>
          </li>
          <li>
            <a
              href="/players/8477341"
            >
              Andreas Johnsson
            </a>
          </li>
          <li>
            <a
              href="/players/8477425"
            >
              Miles Wood
            </a>
          </li>
          <li>
            <a
              href="/players/8477435"
            >
              Ryan Graves
            </a>
          </li>
          <li>
            <a
              href="/players/8477970"
            >
              Vitek Vanecek
            </a>
          </li>
          <li>
            <a
              href="/players/8478399"
            >
              Jonas Siegenthaler
            </a>
          </li>
          <li>
            <a
              href="/players/8478507"
            >
              John Marino
            </a>
          </li>
          <li>
            <a
              href="/players/8479407"
            >
              Jesper Bratt
            </a>
          </li>
          <li>
            <a
              href="/players/8479414"
            >
              Nathan Bastian
            </a>
          </li>
          <li>
            <a
              href="/players/8479415"
            >
              Michael McLeod
            </a>
          </li>
          <li>
            <a
              href="/players/8480002"
            >
              Nico Hischier
            </a>
          </li>
          <li>
            <a
              href="/players/8480003"
            >
              Jesper Boqvist
            </a>
          </li>
          <li>
            <a
              href="/players/8480188"
            >
              Fabian Zetterlund
            </a>
          </li>
          <li>
            <a
              href="/players/8480860"
            >
              Kevin Bahl
            </a>
          </li>
          <li>
            <a
              href="/players/8481033"
            >
              Akira Schmid
            </a>
          </li>
          <li>
            <a
              href="/players/8481068"
            >
              Yegor Sharangovich
            </a>
          </li>
          <li>
            <a
              href="/players/8481559"
            >
              Jack Hughes
            </a>
          </li>
          <li>
            <a
              href="/players/8482110"
            >
              Dawson Mercer
            </a>
          </li>
          <li>
            <a
              href="/players/8482125"
            >
              Alexander Holtz
            </a>
          </li>
        </ul>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
